### PR TITLE
new cursor icon

### PR DESCRIPTION
**Pull request by pandrr**
Screenshot 2021-12-15 at 14 39 58

a text cursor icon

### CODE REVIEW

**Comment by mittalyashu** [comment]
@pandrr Can you please rename it to `text-cursor`?

**Comment by pandrr** [comment]
@mittalyashu done!

**Comment by Andreto** [comment]
@pandrr this path needs to be cleaned up. 
image

And also, try to keep things aligned, evenly spaced, and with proper distance to the edge.
[ICON DESIGN GUIDE](https://github.com/lucide-icons/lucide/blob/master/docs/ICON_DESIGN_GUIDE.md)

**Comment by ericfennis** [comment]
Also the dash in the middle looks a bit weird to me, or is that me?

**Comment by Andreto** [comment]
> Also the dash in the middle looks a bit weird to me, or is that me? @ericfennis 

I use windows and have only ever seen this one: image
But googling "text cursor icon" a lot of the results do have this line.

On Font awesome its called `i-cursor`: image
Is it perhaps a mac thing? I don't know?

Also, something like this could perhaps be useful as a `rename` or `edit-text` icon:
image

**Comment by Andreto** [comment]
imageimage
image

**Comment by pandrr** [comment]
yes on mac there is this little line in the middle

**Comment by ericfennis** [comment]
@pandrr Oh wow didn't see that ! just noticed it.
But I think we need to keep it simple as possible.

**Comment by dudumaluf** [comment]
[cursor_and_rename_icons_24px.zip](https://github.com/lucide-icons/lucide/files/7839780/cursor_and_rename_icons_24px.zip)
Cleaned the cursor icon path according to the guideline and also created the "rename" icon as @Andreto Andreto suggested.

**Comment by pandrr** [comment]
we cleaned up the icons and added a rename.svg as requested.
i hope those are fine now. let me know

**Comment by mittalyashu** [comment]
@pandrr We've a optimise script in `package.json` can you run that on those 2 icons?

File: icons/text-cursor.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" x="0" y="0" viewBox="0 0 24 24" style="enable-background:new 0 0 24 24" xml:space="preserve"><style>.st1{fill:none;stroke:#000;stroke-width:2;stroke-linecap:round;stroke-linejoin:round}</style><path style="fill:none" d="M0 0h24v24H0z" id="Layer_2_1_"/><g id="Layer_1_1_"><path class="st1" d="M7.1 2h1.2c2 0 3.7 1.7 3.7 3.5v12.9c0 2-1.6 3.5-3.5 3.5H7.1M16.9 22h-1.3c-2 0-3.5-1.6-3.5-3.5V5.6C12 3.7 13.7 2 15.7 2h1.2"/></g></svg>
```
Comment (ericfennis) [inline icons/text-cursor.svg:1]:
```suggestion
<svg
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="M17 22h-1a4 4 0 0 1-4-4V6a4 4 0 0 1 4-4h1" />
  <path d="M7 22h1a4 4 0 0 0 4-4v-1" />
  <path d="M7 2h1a4 4 0 0 1 4 4v1" />
</svg>
```

File: icons/rename.svg [hunk 2]
```diff
@@ -0,0 +1 @@
+<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" x="0" y="0" viewBox="0 0 24 24" style="enable-background:new 0 0 24 24" xml:space="preserve"><style>.st1{fill:none;stroke:#000;stroke-width:2;stroke-linecap:round;stroke-linejoin:round}</style><path style="fill:none" d="M0 0h24v24H0z" id="Layer_2_1_"/><g id="Layer_1_1_"><path class="st1" d="M5.3 4h.9C7.7 4 9 5.3 9 6.7v10.5c0 1.6-1.1 2.7-2.7 2.7h-1M12.7 20h-1C10.1 20 9 18.9 9 17.3V6.7C9 5.2 10.2 4 11.8 4h.9"/></g><path class="st1" d="M4.8 16.1h-.7C3 16.1 2 15.1 2 14v-4c0-1.1 1-2.1 2.1-2.1h.7M13.1 7.9h6.8c1.1 0 2.1 1 2.1 2.1v4c0 1.1-.9 2.1-2.1 2.1h-6.8"/></svg>
```
Comment (ericfennis) [inline icons/rename.svg:1]:
```suggestion
<svg
  xmlns="http://www.w3.org/2000/svg"
  width="24"
  height="24"
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <path d="M5.3 4h.9C7.7 4 9 5.3 9 6.7m0 0v10.5c0 1.6-1.1 2.7-2.7 2.7h-1M9 6.7v10.6a2.6 2.6 0 0 0 2.7 2.7h1M9 6.7C9 5.2 10.2 4 11.8 4h.9" />
  <path d="M13.1 7.9h6.8c1.1 0 2.1 1 2.1 2.1v4c0 1.1-.9 2.1-2.1 2.1h-6.8" />
  <path d="M4.8 16.1h-.7C3 16.1 2 15.1 2 14v-4c0-1.1 1-2.1 2.1-2.1h.7" />
</svg>
```

**Review by ericfennis — CHANGES_REQUESTED** [review]
Some code formatting

**Comment by Andreto** [comment]
@pandrr 
Keep separate lines as separate shapes for ease of editing.
This is both on the main cursor and the text box.

The cursor shape should probably also be cut up into three parts to avoid overlapping lines. Something like this:
image

And there is no reason to not align these lines to the grid:
image

**Comment by ericfennis** [comment]
@pandrr @Andreto I fixed that in the formats, see suggestions.

**Comment by pandrr** [comment]
thanks i commited those requests.

**Review by ericfennis — APPROVED** [review]
@pandrr Awesome work!

**Comment by ericfennis** [comment]
@locness3 You ok with the naming of the files?

**Pull request merged**